cmd/virtiofs-dispatcher/main.c

```diff
@@ -0,0 +1,302 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Throwing ideas, how to test this. Can we compile it into a shared library and call dlopen from the golang tests? At least we could test some of the functions defined here